### PR TITLE
Avoid hardcoded conversion to float32 in GraphBuilder

### PR DESCRIPTION
**Pull request by jobrachem**
There were some places in the GraphBuilder that had a hard-coded conversion to float32. I added an init argument that now makes it possible to avoid this behavior completely. The `lsl.Model` likewise gains the `to_float32` boolean flag as an argument, since if `grow=True`, the model internally uses a graph builder instance for growing.

### CODE REVIEW

File: liesel/model/model.py [hunk 1]
```diff
@@ -276,7 +283,7 @@ def _set_missing_names(self) -> GraphBuilder:
         return self
 
     def add(
-        self, *args: Node | Var | GraphBuilder, to_float32: bool = True
+        self, *args: Node | Var | GraphBuilder, to_float32: bool | None = None
```
Comment (wiep) [inline liesel/model/model.py:286]: None option is not explained in doc string.

File: liesel/model/model.py [hunk 2]
```diff
@@ -336,7 +346,9 @@ def add(
 
         return self
 
-    def add_groups(self, *groups: Group, to_float32: bool = True) -> GraphBuilder:
+    def add_groups(
+        self, *groups: Group, to_float32: bool | None = None
```
Comment (wiep) [inline liesel/model/model.py:350]: None option not mentioned in docstring.

**Review by wiep — CHANGES_REQUESTED** [review]
See the two comments about mentioning the None option in the docstrings.

**Comment by jobrachem** [comment]
@wiep thank you! I added the missing information to the docs.

**Comment by jobrachem** [comment]
See https://github.com/liesel-devs/liesel/issues/213 regarding the tests timing out.

**Review by wiep — APPROVED** [review]
perfect; ~~should we wait for #213 before merging?~~

**Pull request merged**